src/services/UserService.ts

```diff
@@ -1,75 +1,27 @@
 import { navigate } from '@reach/router';
-import { BaseService, MethodType } from './BaseService';
-
-export interface IUserProfile {
-  firstName?: string;
-  lastName?: string;
-  user?: {
-    id?: string;
-  };
-}
+import { AppError } from '../Errors';
+import { BaseService } from './BaseService';
+import { ILoginResponse } from './models/user/ILoginResponse';
+import { IUserProfile } from './models/user/IUserProfile';
+import StorageService from './StorageService';
 
 enum Endpoints {
   Edit = 'user/edit',
   Profile = 'user/profile',
   Login = 'user/login',
+  Register = 'user/register',
+  RegisterConfirm = 'user/register/confirm',
 }
 
-export class UserService extends BaseService {
-  private static localStorageTokenID = 'token';
-
-  /**
-   * Getter for user jwt token
-   * @returns token
-   */
-  private static getToken() {
-    return localStorage.getItem(this.localStorageTokenID);
-  }
-
-  /**
-   * Setter for user jwt token
-   * @param token token
-   */
-  private static setToken(token: string) {
-    localStorage.setItem(this.localStorageTokenID, token);
-  }
-
-  /**
-   * Makes an authenticated request to the server, with the token saved in localStorage.
-   * If no token is saved throws an error.
-   * @param method Request method
-   * @param endpoint Request endpoint
-   * @param body Request body
-   * @returns Response from requst
-   */
-  public static async authenticatedRequest(
-    method: MethodType,
-    endpoint: string,
-    body?: Record<string, unknown>
-  ): Promise<Response> {
-    const token = this.getToken();
-    if (!token) {
-      throw new Error('User is not logged in!');
-    }
-
-    return this.request(
-      method,
-      endpoint,
-      {
-        Authorization: `Bearer ${token}`,
-      },
-      body
-    );
-  }
-
+class UserService extends BaseService {
   /**
    * Getter for the profile data of the logged in user.
    * @returns Profile Data
    */
   public static async getProfile(): Promise<IUserProfile> {
     const response = await this.authenticatedRequest('GET', Endpoints.Profile);
 
-    if (!response.ok) throw new Error('Could not get profile!');
+    if (!response.ok) throw new AppError('AUTH_TOKEN_INVALID');
```

Comment:
Change this too

Reply:
Nah this one is ok i think. If the response is not ok then either its a server error or the token was invalid.